openquake/hazardlib/shakemap/parsers.py

```diff
@@ -234,6 +236,173 @@ def local_time_to_time_event(local_time):
     return 'transit'
 
 
+def read_usgs_stations_json(stations_json_str):
+    sj = json.loads(stations_json_str)
+    if 'features' not in sj or not sj['features']:
+        raise LookupError('Station data is not available yet.')
+    stations = pd.json_normalize(sj, 'features')
+    stations['eventid'] = sj['metadata']['eventid']
+    # Rename columns
+    stations.columns = [
+        col.replace('properties.', '') for col in stations.columns]
+    # Extract lon and lat
+    stations[['lon', 'lat']] = pd.DataFrame(
+        stations['geometry.coordinates'].to_list())
+    # Get values for available IMTs (PGA and SA)
+    # ==========================================
+    # The "channels/amplitudes" dictionary contains the values recorded at
+    # the seismic stations. The values could report the 3 components, in such
+    # cases, take the componet with maximum PGA (and in absence of PGA, the
+    # first IM reported).
+    channels = pd.DataFrame(stations.channels.to_list())
+    vals = pd.Series([], dtype='object')
+    for row, rec_station in channels.iterrows():
+        rec_station.dropna(inplace=True)
+        # Iterate over different columns. Each colum can be a component
+        data = []
+        pgas = []
+        for _, chan in rec_station.items():
+            if chan["name"].endswith("Z") or chan["name"].endswith("U"):
+                continue
+            # print(chan["name"])
+            df = pd.DataFrame(chan["amplitudes"])
+            if 'pga' in df.name.unique():
+                pga = df.loc[df['name'] == 'pga', 'value'].values[0]
+            else:
+                pga = df['value'][0]
+            if pga is None or pga == "null":
+                continue
+            if isinstance(pga, str):
+                pga = float(pga)
+            pgas.append(pga)
+            data.append(chan["amplitudes"])
+        # get values for maximum component
+        if pgas:
+            max_componet = pgas.index(max(pgas))
+            vals[row] = data[max_componet]
+        else:
+            vals[row] = None
+    # The "pgm_from_mmi" dictionary contains the values estimated from MMI.
+    # Combine both dictionaries to extract the values.
+    # They are generally mutually exclusive (if mixed, the priority is given
+    # to the station recorded data).
+    try:
+        # Some events might not have macroseismic data, then skip them
+        vals = vals.combine_first(stations['pgm_from_mmi']).apply(pd.Series)
+    except Exception as e:
+        print(e, 'not available in json')
```

Comment:
The print will not be visible from the user. Please decide if we want to ignore this altogether or raise an error. Logging is probably not a good idea because the user normally does not look at the logs and this information should go to the user (the one who has the input file and can fix it) and not to us.